boltstub/tests/test_wiring.py

```diff
@@ -0,0 +1,341 @@
+
+from functools import reduce
+from random import getrandbits
+
+import pytest
+
+from ..wiring import (
+    create_wire,
+    negotiate_socket,
+    RegularSocket,
+    WebSocket,
+)
+
+
+def randbytes(size):
+    return bytearray(getrandbits(8) for _ in range(size))
+
+
+class TestRegularSocket:
+    class TestWhenCacheIsSupplied:
+
+        def test_recv_should_return_cache(self, mocker):
+            cache = b"abc"
+
+            socket_mock = mocker.Mock()
+            regular_socket = RegularSocket(socket_mock, cache)
+
+            received = regular_socket.recv(1024)
+
+            assert received == cache
+            socket_mock.recv.assert_not_called()
+
+        def test_accessing_the_socket_during_second_call(self, mocker):
+            cache = b"abc"
+            expected = b"barcelos"
+
+            socket_mock = mocker.Mock()
+            socket_mock.recv.return_value = expected
+            regular_socket = RegularSocket(socket_mock, cache)
+
+            regular_socket.recv(1024)
+            actual = regular_socket.recv(1024)
+
+            assert actual == expected
+            socket_mock.recv.assert_called_with(1024)
+
+    class TestWhenCacheIsEmpty:
+
+        @pytest.mark.parametrize("empty_cache", [
+            (None),
+            (b"")
```

Comment:
```suggestion
            None,
            b"",
```
No need for parentheses + I'm a fan of trailing commas to reduce git diffs

Reply:
There are more places like this that I will not comment on.